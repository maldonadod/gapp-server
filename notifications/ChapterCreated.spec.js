jest.mock('fcm-push')

const {
  chapter_created_format
  ,chapterCreated
} = require('./index')

const chapter_created_format_mock = jest.fn(chapter_created_format)

const chapter = {
  description: 'The description',
  author: {
    first_name: 'Pepe'
  },
  guests: [
    {
      user: {
        regid: 'regid1'
      }
    }
    ,{
      user: {
        regid: 'regid2'
      }
    }
  ]
}

describe('ChapterCreated', () => {
  
  test('chapter created payload should match', () => {
    expect(chapter_created_format_mock(chapter)).toMatchSnapshot()
  })
  
  test('send response match with snapshot', () => {
    expect.assertions(1)
    chapterCreated(chapter)
    .then(res => {
      expect(res).toMatchSnapshot()
    })
  })
})