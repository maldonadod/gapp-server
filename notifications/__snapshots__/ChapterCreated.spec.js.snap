// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterCreated chapter created payload should match 1`] = `
Array [
  "Pepe invite you",
  Array [
    "regid1",
    "regid2",
  ],
]
`;

exports[`ChapterCreated chapter created payload should match 2`] = `
Object {
  "collapse_key": undefined,
  "data": "Pepe invite you",
  "notification": undefined,
  "to": Array [
    "regid1",
    "regid2",
  ],
}
`;
